wayland-egl/src/lib.rs

```diff
@@ -40,17 +41,17 @@ impl WlEglSurface {
     ///
     /// This method will check that the provided `ObjectId` is still alive and from the
     /// correct interface (`wl_surface`).
-    pub fn new(surface: ObjectId, width: i32, height: i32) -> Result<WlEglSurface, InvalidId> {
+    pub fn new(surface: ObjectId, width: i32, height: i32) -> Result<WlEglSurface, Error> {
         if surface.interface().name != "wl_surface" {
-            return Err(InvalidId);
+            return Err(Error::InvalidId);
         }
 
         let ptr = surface.as_ptr();
         if ptr.is_null() {
-            Err(InvalidId)
+            Err(Error::NullPointer)
```

Comment:
I think the two checks should be fused into a single if (using `ObjectId::is_null()`), and the `Error::NullPointer` variant can be removed:

```rust
if surface.is_null() || surface.interface().name != "wl_surface" {
    return Err(Error::InvalidId)
}
```